app/controllers/api/v1/callbacks_controller.rb

```diff
@@ -19,31 +19,42 @@ def get_facebook_pages
     @page_details = mark_already_existing_facebook_pages(fb_object.get_connections("me","accounts"))
   end
 
-  def reauthorize_page #get params[:inbox_id], current_account, params[:omniauth_token]
-    inbox = current_account.inboxes.find_by(id: params[:inbox_id])
-    if inbox
-      fb_page_id = inbox.channel.page_id
+  # get params[:inbox_id], current_account, params[:omniauth_token]
+  def reauthorize_page
+    if @inbox
```

Comment:
the `@inbox`  is defined in the `inbox` method.  and I don't see the method being called anywhere before the `if` 
I assume it resolves to nil always? 

we should use `inbox` here instead?

Reply:
yes you're right. i did not see any test so i did not add. i can add some test for that case.

Reply:
@enderahmetyurt  We didn't get a chance to add tests when we wrote the initial version. But now that we are getting contributions, I think the tests would help us to move fast without breaking stuff.  
So these will be the initial steps in that direction. thanks a lot for the help.

Reply:
when I was writing test for `reauthorize_page` method. I got an error `undefined method `page_id' for #<Channel:0x00007f9e66fb0070>` I looked to `channel.rb` model and i cannot see any `page_id` method. Could you help me? How should I get it? Cheers.

Reply:
@enderahmetyurt  the `page_id` was moved to `facebook_page.rb` at some point and whoever worked on the code here forgot to update this part I guess. 🤦🏻‍♀️ cause we don't have tests.  

for a short history, we initially wrote the tool just to support facebook pages. then decided to add multi-channel support.  while refactoring the models and relations I believe this part was messed up.   

we should add an `inbox_id` attribute to the `facebook_pages` model.  and in the `register_facebook_page` in 'callbacks_controller.rb` ensure that the inbox_id is populated while facebook_page is created. 'ie: line 13' [move the inbox_creation in line 14 to to above this]

now in the `reauthorize_page` method, we will be able to fetch the page_id by querying the facebook_pages model with the inbox id.  

so overall we will have to do a bit of refactoring here, if you think you have the time to tackle this let me know, or i will fix up this part and pass this back to you.  





```
fb_page_id = facebook_page.

Reply:
added: https://github.com/chatwoot/chatwoot/issues/143  to tackle this issue.

Reply:
hmm I got it @sojan-official I don't have enough time to fix it. Could you fix up this part? After that I can go on.

Reply:
@enderahmetyurt  sure no worries. We will fix up that part and pass it back to you.

Reply:
so @sojan-official how is it going? Is there any improvements about that issue?

Reply:
@enderahmetyurt  sorry, i had lost track on this PR. we also made some changes over how the channel models are structured. Let me test this branch based on that in the next couple of days and let's get this merged.  

sorry for the lack of updates again.